backend/tools/keycloak/sync/Exceptions/HttpResponseException.cs

```diff
@@ -0,0 +1,35 @@
+using Pims.Tools.Keycloak.Sync.Extensions;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json;
+
+namespace Pims.Tools.Keycloak.Sync.Exceptions
+{
+    public class HttpResponseException : Exception
+    {
+        #region Properties
+        public string Details { get; }
+        public HttpStatusCode StatusCode { get; }
+        #endregion
+
+        #region Constructors
+        public HttpResponseException(HttpResponseMessage response, string message = null) : base(message)
+        {
+            this.StatusCode = response.StatusCode;
+            using var stream = response.Content.ReadAsStreamAsync().Result;
+            if (response.Content.Headers.ContentType == new MediaTypeHeaderValue("application/json"))
+            {
+                var results = JsonSerializer.DeserializeAsync<object>(stream).Result;
+                this.Details = JsonSerializer.Serialize(results); // TODO: Not ideal to return JSON as the error.
```

Comment:
is it weird that we are deserializing the json stream and then serializing it again right afterwords?

Reply:
It is weird.  It was a quick hack for me to copy paste some code to get the string value out of the steam.  It should just read the stream though.

Reply:
A now I remember what this originated from.  It used to deserialize the JSON object to extract a value from it.  I'll change it to clean it up.